src/rust/engine/fs/src/store.rs

```diff
@@ -591,6 +591,18 @@ impl Store {
         RootOrParentMetadataBuilder::Root(root.clone()),
         digest,
       )
+      .and_then(|files| {
```

Comment:
Also, given the reintroduction of non-blocking IO in here, how will we confirm that this `sync_all` call will be happening in a non-blocking fashion? By using a tokio-specific `sync_all`...?

Reply:
Well, there is also blocking dir creation happening via calls to:
https://github.com/pantsbuild/pants/blob/6890cc098e612876131fe2e76f1d2bd8d093b35e/src/rust/engine/fs/src/lib.rs#L790-L821

From:
https://github.com/pantsbuild/pants/blob/b11904dfbe3c47e46ed32a1da3ec986c1a958fbf/src/rust/engine/fs/store/src/lib.rs#L620-L630

That should probably be fixed too.